src/modules/common/types/api-types.d.ts

```diff
@@ -63,4 +63,32 @@ export interface IChapterInfoAPIResponse {
     seoDescription: string;
     chapTitle: string;
     checkVol2Chap1: boolean;
+}
+
+export interface ISearchAPIResponse {
+    id: number;
+    hid: string;
+    slug: string;
+    title: string;
+    rating: number | null;
+    bayesian_rating: number | null;
+    rating_count: number;
+    follow_count: number;
+    desc: string;
+    status: number;
+    last_chapter: number;
+    translation_completed: boolean;
+    view_count: number;
+    content_rating: string;
+    demographic: number | null;
+    uploaded_at: string;
+    genres: number[];
+    created_at: string;
+    user_follow_count: number;
+    year: number;
+    country: string;
+    md_titles: { title: string };
+    md_covers: IMDCover[];
+    mu_comics: { year: number };
+
 }
```

Comment:
This is just a suggestion, but you could possibly have optional values where the type can be either specified or null. For example, have `rating?: number;` instead of `rating: number | null;`. This is subject to your preference.